caikit/interfaces/runtime/server.py

```diff
@@ -0,0 +1,93 @@
+# Copyright The Caikit Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+This file contains interfaces and descriptor functions for the runtime servers
+"""
+
+# Standard
+from enum import Enum
+import re
+
+# Local
+from caikit.config import get_config
+from caikit.interfaces.runtime.service import (
+    ServiceType,
+    get_service_name,
+    get_service_package_name,
+)
+
+### HTTP Server
+
+# Endpoint to use for health checks
+HEALTH_ENDPOINT = "/health"
+
+# Endpoint to use for server info
+RUNTIME_INFO_ENDPOINT = "/info/version"
+MODELS_INFO_ENDPOINT = "/info/models"
+
+# These keys are used to define the logical sections of the request and response
+# data structures.
+REQUIRED_INPUTS_KEY = "inputs"
+OPTIONAL_INPUTS_KEY = "parameters"
+MODEL_ID = "model_id"
+
+# Stream event types enum
+class StreamEventTypes(Enum):
+    MESSAGE = "message"
+    ERROR = "error"
+
+
+def get_http_route_name(rpc_name: str) -> str:
+    """Function to get the http route for a given rpc name
+
+    Args:
+        rpc_name (str): The name of the Caikit RPC
+
+    Raises:
+        NotImplementedError: If the RPC is not a Train or Predict RPC
+
+    Returns:
+        str: The name of the http route for RPC
+    """
+    if rpc_name.endswith("Predict"):
```

Comment:
I'm assuming this is just ported, but I don't _love_ how brittle all of this name convention stuff is

Reply:
Yeah, I had a similar thought but wasn't sure how to best refactor it.